rwengine/src/items/Weapon.hpp

```diff
@@ -1,12 +1,64 @@
 #ifndef _RWENGINE_WEAPON_HPP_
 #define _RWENGINE_WEAPON_HPP_
+#include <glm/glm.hpp>
 
 class CharacterObject;
+class GameObject;
 struct WeaponData;
 
+/**
+ * @brief simple object for performing weapon checks against the world
+ */
+struct WeaponScan {
+    enum ScanType {
+        /** Instant-hit ray weapons */
+        HITSCAN,
+        /** Area of effect attack */
+        RADIUS,
+    };
+
+    const ScanType type;
+
+    float damage;
+
+    glm::vec3 center{};
+    float radius;
+
+    glm::vec3 end{};
+
+    WeaponData* weapon;
+    GameObject* source;
+
+    // Constructor for a RADIUS hitscan
+    WeaponScan(float damage, const glm::vec3& center, float radius,
+               WeaponData* weapon = nullptr, GameObject* source = nullptr)
+            : type(RADIUS)
```

Comment:
Maybe extracting `ScanType` from `WeaponScan` and/or making it enum class will increase readability?
`type(ScanType::RADIUS)`

Reply:
btw it looks like macro :p

Reply:
Yes this is quite old stuff. Since the PR is quite small still I'll tidy up the things you've flagged.